crypto/src/lib.rs

```diff
@@ -110,66 +111,73 @@ impl FromStr for Algorithm {
     }
 }
 
-/// Options for key generation
-#[cfg(not(feature = "ffi_import"))]
-#[derive(Debug, Clone)]
-pub enum KeyGenOption<K = PrivateKey> {
+/// Needed since it couldn't be represented in FFI
+#[derive(Clone)]
+#[cfg_attr(not(feature = "ffi_import"), derive(Debug))]
+enum KeyPairGenConfigInner {
+    /// Use random number generator
+    #[cfg(feature = "rand")]
+    Random { algorithm: Algorithm },
+    /// Use seed
+    UseSeed { seed: Vec<u8>, algorithm: Algorithm },
+    /// Derive from a private key
+    FromPrivateKey(PrivateKey),
+}
+
+/// While [`KeyPairGenConfig`] is for external users, this structure is for internal use
+/// to pass it to a specific algorithm.
+/// Parameter `K` is intended to be set as a `PrivateKey` type of that specific algorithm.
+#[derive(Debug)]
+pub enum KeyPairGenOption<K> {
     /// Use random number generator
     #[cfg(feature = "rand")]
     Random,
     /// Use seed
     UseSeed(Vec<u8>),
-    /// Derive from private key
+    /// Derive from a private key
     FromPrivateKey(K),
 }
 
 ffi::ffi_item! {
-    /// Configuration of key generation
+    /// Configuration for key generation
     #[derive(Clone)]
-    #[cfg_attr(not(feature="ffi_import"), derive(Debug))]
-    pub struct KeyGenConfiguration {
-        /// Options
-        key_gen_option: KeyGenOption,
-        /// Algorithm
-        algorithm: Algorithm,
+    #[cfg_attr(not(feature = "ffi_import"), derive(Debug))]
+    pub struct KeyPairGenConfig{
```

Comment:
```suggestion
    pub struct KeyGenConfig{
```

I think previous name was fine